### DISCUSSION TITLE
Comparison with Nitro

### DISCUSSION THREAD

**Discussion by brillout** [hattipjs/hattip#169] [General]
Nitro is getting a lot of hype. I'm thinking HatTip could ride on Nitro's wave by telling users how it compares itself with Nitro.

**Comment by redbar0n** [comment]
FWIW

> Nitro is the universal backend that powers a lot of the magic in Nuxt. You can think of it as everything except for the rendering layer of your application (everything except for Vue).
>
> h3 is the HTTP framework that powers Nitro. Nitro layers on file-based routing, asset handling, storage abstraction, and many more features to turn it into a very powerful framework.

https://masteringnuxt.com/nuxt3/lessons/5-16-understanding-nitro

So to me it seems like HatTip would compare to H3.

Nitro would compare to Vike, I guess. Although I couldn't figure out if Nitro is based on Vite or not (but Nuxt 3 is, so presumably yes... but there is a [vite-plugin-nitro](https://www.npmjs.com/package/@analogjs/vite-plugin-nitro) needed...). Furthermore, Nitro is focused on API endpoints, whereas [Vike isn't](https://vike.dev/api-routes).

**Comment by redbar0n** [comment]
Maybe Nitro is not a substitute for Vike, since here's an example using Nitro with Vike...:

[turbo-nitro-vike-example](https://github.com/ThimoDEV/turbo-nitro-vike-example)

Though I came across this...:

[Servite](https://github.com/Codpoe/servite) - "A vite plugin for React SSR / SSG / CSR / Islands. Powered by nitro."

Apparently, Vite plugs into Nitro, which is the web server framework (on top of H3):

"Vite actually fits into one of the routing handlers into Nitro. Nitro owns the server both during development and production, and then Vite kind of plugs in. Like so for example the Static Server doesn't need Vite, but the client servers do." -- [according to](https://youtu.be/WBH7Ewa--Qg?t=2749) Nikhil Saraf, author of Vinxi (built on top of Nitro)
